server.go

```diff
@@ -177,3 +181,19 @@ func (s *Server) DeserializeKE2(ke2 []byte) (*message.KE2, error) {
 func (s *Server) DeserializeKE3(ke3 []byte) (*message.KE3, error) {
 	return s.Parameters.DeserializeKE3(ke3)
 }
+
+// DeserializeAKEState sets the interal state of the AKE server from the given
+// bytes
+func (s *Server) DeserializeAKEState(state []byte) error {
+	if len(state) != s.MAC.Size() + s.KDF.Size() {
+		return errors.New("state is the wrong length")
+	}
+
+	return s.Ake.SetState(state[:l], state[l:])
```

Comment:
```suggestion
	return s.Ake.SetState(state[:s.MAC.Size()], state[s.KDF.Size():])
```

Reply:
Wouldn't it be like this?

```suggestion
	return s.Ake.SetState(state[:s.MAC.Size()], state[s.MAC.Size():s.MAC.Size()+s.KDF.Size()])
```

or more simply:

```suggestion
	return s.Ake.SetState(state[:s.MAC.Size()], state[s.MAC.Size():])
```

Reply:
Yes, you're right! Let's go for the simpler variant 👍